grakn-graph/src/main/java/ai/grakn/graph/internal/RoleTypeImpl.java

```diff
@@ -91,7 +91,7 @@
      */
     public Set<CastingImpl> castings(){
         Set<CastingImpl> castings = new HashSet<>();
-        getIncomingNeighbours(Schema.EdgeLabel.ISA).forEach(concept -> ((CastingImpl) concept).getRelations().forEach(relation -> getGraknGraph().getConceptLog().putConcept(relation)));
+        getIncomingNeighbours(Schema.EdgeLabel.ISA).forEach(concept -> castings.add((CastingImpl) concept));
```

Comment:
But @fppt it's cold here in the UK! Sometimes the laptop is the only thing keeping me warm...